nixos/modules/services/web-apps/librechat.nix

```diff
@@ -0,0 +1,149 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.librechat;
+  configFile = pkgs.writeTextFile {
+    name = "librechat.yaml";
+    text = lib.generators.toYAML { } cfg.settings;
+  };
+  # Thanks to https://github.com/nix-community/home-manager/blob/60e4624302d956fe94d3f7d96a560d14d70591b9/modules/lib/shell.nix :)
+  export = n: v: ''export ${n}="${builtins.toString v}"'';
+  exportAll = vars: lib.concatStringsSep "\n" (lib.mapAttrsToList export vars);
+  exportCredentials = n: _: ''export ${n}="$(${pkgs.systemd}/bin/systemd-creds cat ${n}_FILE)"'';
+  exportAllCredentials = vars: lib.concatStringsSep "\n" (lib.mapAttrsToList exportCredentials vars);
+  transformCredential = n: v: "${n}_FILE:${v}";
+  getLoadCredentialList = lib.mapAttrsToList transformCredential cfg.credentials;
+  allowedPorts = lib.optionals cfg.openFirewall (
+    if cfg.env ? PORT then [ cfg.env.port ] else [ 3080 ]
+  );
```

Comment:
This doesn't look ideal. For vital configuration options like these we should offer a top level option with type `lib.types.port`